### ISSUE TITLE
Possible to use with styled-system?

### ISSUE THREAD

**Issue by DanielTamkin** [santosfrancisco/react-awesome-styled-grid#107]
Been playing around with this library, it's fantastic @santosfrancisco! But i would love to combine it with [theme-ui](https://theme-ui.com). _Why?_ Because i use [rebassjs](https://rebassjs.org/theming) to quickly build out some designs. Using the styled option so i don't have to utilize the bloated emotion library. 

Has anyone successfully implemented this library with theme-ui? 

The main problem is theme-ui does not use the default styled-components `ThemeProvider`, therefore breaking this library.

**Comment by DanielTamkin** [comment]
If this is not possible for the current system, will spend time to PR a fix for this.

**Comment by DanielTamkin** [comment]
For anyone else that comes across this edge-case:

Define a specific themeprovider for just awesomegrid. I was incorrectly grouping it with theme-ui's config object. Causing this library to never get the context needed.

So instead of 
```javascript
import { ThemeProvider } from 'theme-ui'
...
<ThemeProvider theme={theme}>
  {...stuff}
</ThemeProvider>
...
```

We will do:
```javascript
import { ThemeProvider } from 'theme-ui'
...
<GridProvider theme={{ awesomegrid: theme.awesomegrid }}>
  <ThemeProvider theme={theme}>
    {...stuff}
  </ThemeProvider>
</GridProvider>
...
```

Hope this catches anyone that comes into a similar usecase!

**Comment by santosfrancisco** [comment]
Hi @DanielTamkin thanks! I don't know if I understood it correctly, but it seems to me that you already solved the problem by importing the `ThemeProvider` from styled-components and passing the grid config, right?

**Comment by DanielTamkin** [comment]
That's correct, curious as to why i have to do this workaround. It's fair to assume that theme-ui handles their `ThemeProvider` Logic different than styled-components, so that would be my guess. Just figured documenting so others might catch this on a search! 

Eventually i'd love to dig into this library and figure out how it is getting it's theme config. Might need an additional bit of logic to handle [theme-spec](https://theme-ui.com/theme-spec).
